workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -398,19 +433,23 @@ def _apply_postphysics_to_dycore_state(self) -> Diagnostics:
                 updated_state_from_tendency = add_tendency(
                     self._state, tendency, dt=self._timestep
                 )
+
+                # if total precip is updated directly by stepper,
```

Comment:
OK I think I follow the logic that makes sure that that total_precipitation updates from state updates are set directly while updates from dQ2 use precipitation_sum. 

Not for this PR but it keeps coming up how useful it would be to have a loop unit test which ensure that all of the rules we've built into the loop are actually followed.

Reply:
> which ensure that all of the rules we've built into the loop are actually followed

The regression tests are pretty good and take maybe 30 seconds to run per configuration.  IMO loop.py is not really unit-testable outside of the regression tests since it interacts so much with the wrapper. 

 I'd rather suggest moving "all of the rules we've built into the loop" into units. The regression tests and mypy are pretty useful for that kind of refactor.

Reply:
For example, the coupling between add_tendency, loop, and stepper could be resolved by having the steppers return a function that applies their internal predicted tendencies to the state. Any physicsy code can be pushed into a stepper. or something like `monitor`.

Reply:
Makes sense. I'd be interested in working on those kinds of refactors at some point.